ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java

```diff
@@ -641,7 +641,8 @@ private static void loadFunctionResourcesIfNecessary(String functionName, Common
       // First try without qualifiers - would resolve builtin/temp functions.
       // Otherwise try qualifying with current db name.
       functionInfo =  mFunctions.get(functionName);
-      if (functionInfo == null && !FunctionUtils.isQualifiedFunctionName(functionName)) {
+      if (functionInfo == null &&
+          SessionState.get() != null && SessionState.get().getCurrentDatabase() != null) {
```

Comment:
This is certainly safe, but is the null check on getCurrentDatabase ever useful?  As implemented, SessionState.getCurrentDatabase() just serves to turn a null currentDatabase into DEFAULT_DATABASE_NAME.